docs/rfcs/20220209_leader_epoch_support_rfc.md

```diff
@@ -0,0 +1,114 @@
+- Feature Name: Leader epoch handling in redpanda
+- Status: in-progress
+- Start Date: 2022-02-09
+- Authors: Michal Maslanka
+- Issue: https://github.com/redpanda-data/redpanda/issues/3689
+
+# Executive Summary
+
+In kafka some of the client Requests are fenced with the Leader Epoch. 
+Leader Epoch is a 32-bit integer that is increased every time partition 
+leadership changes. The Leader Epoch is used to detect stale metadata, 
+fence produce and fetch request and recognize truncations.
+
+## What is being proposed
+
+Using Raft protocol `term` as a Kafka leader epoch. The `term` is semantically 
```

Comment:
curious to see how kraft is thinking about this.

Reply:
Exactly the same, in raft implementation they do not even use the word `term`